great_expectations/cli/suite.py

```diff
@@ -117,6 +120,9 @@ def suite_new(
     context: DataContext = ctx.obj.data_context
     usage_event_end: str = ctx.obj.usage_event_end
 
+    # Only set to true if `--profile` or `--profile <PROFILER_NAME>`
+    profile: bool = profiler_name is not None
+
```

Comment:
Perhaps we could leverage a single variable but I found a `profile` variable that acts as a boolean switch and `profiler_name` identifier easier to reason about.

This maintains the underlying structure noted in the various utility methods and seemed to be the least disruptive way to implement.